orchagent/pfcwdorch.cpp

```diff
@@ -939,26 +949,40 @@ void PfcWdSwOrch<DropHandler, ForwardHandler>::doTask(SelectableTimer &timer)
 
 template <typename DropHandler, typename ForwardHandler>
 void PfcWdSwOrch<DropHandler, ForwardHandler>::report_pfc_storm(
-        sai_object_id_t id, const PfcWdQueueEntry *entry)
+        sai_object_id_t id, const PfcWdQueueEntry *entry, const string &info)
 {
     event_params_t params = {
         { "ifname", entry->portAlias },
         { "queue_index", to_string(entry->index) },
         { "queue_id", to_string(id) },
         { "port_id", to_string(entry->portId) }};
 
-    SWSS_LOG_NOTICE(
-            "PFC Watchdog detected PFC storm on port %s, queue index %d, queue id 0x%" PRIx64 " and port id 0x%" PRIx64 ".",
+    if (info.empty())
+    {
+        SWSS_LOG_NOTICE(
+            "PFC Watchdog detected PFC storm on port %s, queue index %d, queue id 0x%" PRIx64 " and port id 0x%" PRIx64,
             entry->portAlias.c_str(),
             entry->index,
             id,
             entry->portId);
+    }
+    else
+    {
+        SWSS_LOG_NOTICE(
+            "PFC Watchdog detected PFC storm on port %s, queue index %d, queue id 0x%" PRIx64 " and port id 0x%" PRIx64 ", additional info: %s.",
+            entry->portAlias.c_str(),
+            entry->index,
+            id,
+            entry->portId,
+            info.c_str());
+        params["additional_info"] = info;
```

Comment:
@zbud-msft , would there be any issues with this additional info getting added?

Reply:
Yes, we will need to make necessary revisions to the sonic-events-swss yang model and the appropriate UT: https://github.com/sonic-net/sonic-buildimage/blob/master/src/sonic-yang-models/yang-models/sonic-events-swss.yang#L55. It is necessary for each event to conform to the YANG model as that is the ground truth.

Reply:
Can you please create build image PR for the YANG model change?

Reply:
https://github.com/sonic-net/sonic-buildimage/pull/16206
It's in draft because it is still being tested internally.

Reply:
Without YANG model PR being merged in, submodule update containing this commit will fail PR checker since pfc storm event would be sending info that is not known.

Reply:
The yang model PR has been merged.